logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/log/Sequencer.java

```diff
@@ -130,7 +86,11 @@ public long tryWrite(
     lock.lock();
     try {
       currentPosition = position;
-      isEnqueued = queue.offer(new SequencedBatch(currentPosition, sourcePosition, entries));
+      //noinspection unchecked
+      isEnqueued =
+          queue.offer(
+              new SequencedBatch(
+                  currentPosition, sourcePosition, (List<LogAppendEntry>) appendEntries));
```

Comment:
:upside_down_face: We could also store it as `List<? extends LogAppendEntry>` in the batch, since I don't think it has to be propagated much, only used in the serializer? Plus we want to make it self-serializable, so it'll be completely unexposed soon :shrug: We can alternatively remove the cast at this point as well